driver/razeraccessory_driver.c

```diff
@@ -791,7 +823,20 @@ static int razer_accessory_probe(struct hid_device *hdev, const struct hid_devic
     // Init data
     razer_accessory_init(dev, intf, hdev);
 
-    if(dev->usb_interface_protocol == USB_INTERFACE_PROTOCOL_MOUSE) {
+    switch(usb_dev->descriptor.idProduct) {
+    case USB_DEVICE_ID_RAZER_CHROMA_MUG:
+    case USB_DEVICE_ID_RAZER_CHROMA_HDK:
+    case USB_DEVICE_ID_RAZER_CHROMA_BASE:
+        expected_protocol = USB_INTERFACE_PROTOCOL_MOUSE;
+        break;
+
+    case USB_DEVICE_ID_RAZER_NOMMO_PRO:
+    case USB_DEVICE_ID_RAZER_NOMMO_CHROMA:
+        expected_protocol = USB_INTERFACE_PROTOCOL_KEYBOARD;
+        break;
+    }
```

Comment:
Please make one of those (probably the _MOUSE) the default case (you can keep the other case statements there, just add a default) otherwise `expected_protocol` will still have the value `0` for devices not added here. Another way would be to set the variable in line 809 directly to e.g. `USB_INTERFACE_PROTOCOL_MOUSE` so we have a sane default value

Reply:
Set expected_protocol initial value to USB_INTERFACE_PROTOCOL_MOUSE in both instances.